### PR TITLE
cmd/syncthing: Warn when running as a super user (fixes #4123)

### LINKED ISSUE #4123: Warn against running as a privileged user
We should generally not run as a privileged user. We can't suddenly make that an error as that would break a lot of installations. But we could add a warning with similar behavior as the already existing one for unprotected GUI access. With a similar advanced config to tone it down. 

Practically, detecting uid 0 is trivial. I don't know how to detect SYSTEM on Windows.

### PR DESCRIPTION
**Pull request by calmh**
UID 0 on Unixes, SYSTEM SID on Windows.

### Testing

I've verified that this works on Unixes. **I'd like help to verify this is the right thing on Windows. Also, are there other SIDs we should be looking for?**

### Screenshot

screen shot 2017-05-14 at 18 12 53

### CODE REVIEW

**Comment by ProactiveServices** [comment]
Syncthing v0.14.28-rc.3+13-gd25b1526 Windows 7 x64, running as NT AUTHORITY\SYSTEM SID S-1-5-18 (both the helper syncthing.exe and "actual" syncthing.exe) is not causing the warning. Nothing in the Syncthing logs to suggest any warning either.

**Comment by calmh** [comment]
Ugh. Can you run with the latest update, and `STTRACE=main`?

**Comment by calmh** [comment]
Oh, ah, `gd25b1526` doesn't have this feature. This is the one to test: https://build.syncthing.net/job/syncthing-pr-windows/2398/

File: lib/protocol/bep.proto [hunk 1]
```diff
@@ -108,7 +108,6 @@ message FileInfo {
 
     repeated BlockInfo Blocks         = 16 [(gogoproto.nullable) = false];
     string             symlink_target = 17;
-    bytes encryption_iv = 18;
```
Comment (AudriusButkevicius) [inline lib/protocol/bep.proto]: hmm

Comment (AudriusButkevicius) [inline lib/protocol/bep.proto]: I still think this is best done via a custom filesystem translator ;)

**Comment by ProactiveServices** [comment]
Warning is present with the linked exe, both in GUI and console.

**Comment by calmh** [comment]
\o/

**Comment by calmh** [comment]
@st-review merge it, it's been tested

**Comment by st-review** [comment]
:ok_hand: Merged as 1d15b8be9bfcd1e7e4063b3169e7d8bd727b7450. Thanks, @calmh!

**Pull request closed without merging**